include/fiction/algorithms/simulation/sidb/displacement_robustness_domain.hpp

```diff
@@ -0,0 +1,670 @@
+//
+// Created by Jan Drewniok on 21.03.24.
+//
+
+#ifndef FICTION_DISPLACEMENT_ROBUSTNESS_DOMAIN_HPP
+#define FICTION_DISPLACEMENT_ROBUSTNESS_DOMAIN_HPP
+
+#include "fiction/algorithms/simulation/sidb/is_operational.hpp"
+#include "fiction/layouts/coordinates.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/utils/layout_utils.hpp"
+#include "fiction/utils/math_utils.hpp"
+
+#include <fmt/format.h>
+#include <mockturtle/utils/stopwatch.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <cstdlib>
+#include <limits>
+#include <mutex>
+#include <random>
+#include <set>
+#include <thread>
+#include <utility>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * During fabrication, SiDBs may not align precisely with their intended atomic positions, resulting in displacement.
+ * This means that an SiDB is fabricated close to the desired one, typically one or a few H-Si
+ * positions away. Consequently, depending on the fabrication speed, a certain number of SiDBs may experience
+ * displacement. To address and analyze this occurrence, we introduce the *Displacement Robustness Domain*. This domain
+ * consists of SiDB layouts derived from an original layout, each showing displaced SiDBs, together with the
+ * `operational` or `non-operational` status, based on the specified logic.
+ */
+template <typename Lyt>
+struct displacement_robustness_domain
+{
+    /**
+     * Represents a domain of displacement robustness for layouts resulting from applying a displacement to a given SiDB
+     * layout.
+     */
+    std::vector<std::pair<Lyt, operational_status>> operational_values{};
+};
+
+/**
+ * Specifies the allowed displacement range options for SiDB placement.
+ */
+enum class dimer_displacement_policy
```

Comment:
What is the difference in importance of this `enum class` compared to `displacement_robustness_domain_params`? The latter has been integrated into the params `struct`, but this one hasn't.

Reply:
you are right! We should put both in params. Thanks!